src/Extensions.Hosting.AsyncInitialization/Hosting/AsyncInitializationHostExtensions.cs

```diff
@@ -32,5 +32,35 @@ public static async Task InitAsync(this IHost host, CancellationToken cancellati
 
             await rootInitializer.InitializeAsync(cancellationToken);
         }
+
+
+        /// <summary>
+        /// Initializes and runs the application, by first calling all registered async initializers.
+        /// Upon completion, any registered initializers implementing <see cref="IAsyncTeardown"/> are called.
+        /// </summary>
+        /// <param name="host">The host.</param>
+        /// <param name="cancellationToken">Optionally propagates notifications that the operation should be cancelled</param>
+        /// <returns>A task that represents the initialization completion.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="InvalidOperationException"></exception>
+        /// <exception cref="OperationCanceledException"></exception>
```

Comment:
If you're not going to indicate _when_ these exceptions are thrown, I think it's not worth mentioning them.

Reply:
I get your point and added a description. But keep in mind that VS currently does not support showing the exception description in IntelliSense (at least to my knowledge - and the description does not show up).
Done.